op-acceptance-tests: various improvements to the Fusaka acceptance test

joshklop opened the pull request:
This is a follow-up to https://github.com/ethereum-optimism/optimism/pull/17529, which merged with nits due to time pressure.

joshklop committed:
split out helpers and initialization to a separate file

joshklop committed:
fix comment in mise.toml

joshklop committed:
rename osaka test

joshklop committed:
rename to fusaka

joshklop committed:
txplan: support non-cell proof blob txs

joshklop committed:
also verify the l1 blob base fee in the l1 block contract

joshklop committed:
optimization

https://github.com/ethereum-optimism/optimism/pull/17529#discussion_r2406241699

joshklop commented:
Hopefully this is straightforward enough to review quickly. I'm leaving two comments on the previous PR for separate follow-ups:

- Being more careful with the blob base fee formula: https://github.com/ethereum-optimism/optimism/pull/17529#discussion_r2405498871
- Separating out the load test code https://github.com/ethereum-optimism/optimism/pull/17529#discussion_r2405521316

geoknee commented on op-acceptance-tests/tests/fusaka/fusaka_test.go:
```diff
@@ -1,120 +1,42 @@
+	l2BlockTime := time.Duration(sys.L2Chain.Escape().RollupConfig().BlockTime) * time.Second
+	for range time.Tick(l2BlockTime) {
+		l2SafeRef := sys.L2EL.BlockRefByLabel(eth.Safe)
+		if l1Config.IsOsaka(new(big.Int).SetUint64(l2SafeRef.Number), l2SafeRef.Time) {
```
Perhaps a test log here to say "chain progresses safe head past osaka activation time stamp".

geoknee commented on op-acceptance-tests/tests/fusaka/init_test.go:
```diff
@@ -0,0 +1,79 @@
+		}),
+		sysgo.WithBatcherOption(func(_ stack.L2BatcherID, cfg *batcher.CLIConfig) {
+			cfg.DataAvailabilityType = flags.BlobsType
+			cfg.TxMgrConfig.CellProofTime = 0 // Force cell proofs to be used
```
I was going to suggest setting this to the l1 osaka time, but it doesn't seem straightforward / possible to compute that here.

joshklop replied:
We could add it here https://github.com/ethereum-optimism/optimism/blob/fd576a8e271a932252b6ee6fd6079da9a316fa64/op-devstack/sysgo/l2_batcher.go#L100

I think we might have to make that change. Am I missing anything?

joshklop committed:
op-acceptance-tests: harden the BPO blob base fee calculation test (#17953)

Before, we merely waited for the blob base fee to rise above 1 before assuming
that the blob parameters would have a visible impact on the fee.

Due to rounding, it is possible that different parameters still result in the same
base fee even above the minimum (either 1 or the reserve price set by EIP-7918).

Calculating the precise intersections between different blob base fee formulae is
complicated, so we opt for the simpler but maximally robust approach: calculate
both the BPO1 base fee and the Osaka base fee, and only proceed when they differ.

This resolves a nit on a previous PR that was merged due to time pressure:
https://github.com/ethereum-optimism/optimism/pull/17529#discussion_r2405498871

joshklop committed:
add log
